### ISSUE TITLE
Callgraph taking a while to build

### ISSUE THREAD

**Issue by will-leeson** [soot-oss/SootUp#558]
I am building a tool that will consume graph representations of programs to perform a task. SootUp has been a great tool for this project. I want to be able to build call graphs for arbitrary functions in a library. I created the following function to build Call Graphs for every method in a given project:

```java
public void buildCallGraphs(){
    for(JavaSootClass aClass : this.localView.getClasses()){
        for(SootMethod aMethod : aClass.getMethods()){
            System.out.println(aMethod.toString());
            MethodSignature entryMethodSignature = aMethod.getSignature();
            CallGraphAlgorithm cha = new ClassHierarchyAnalysisAlgorithm(globalView, typeHierarchy);

            CallGraph cg = cha.initialize(Collections.singletonList(entryMethodSignature));

            System.out.println("CALLS");
            for(MethodSignature m:cg.callsFrom(entryMethodSignature)){
                System.out.println(m.getName());
            }
            System.out.println("CALLS END");
        }
    }
}
``` 

When I run this code, some iterations finish almost instantly, while others take over 10 minutes. I understand building the call graph can be expensive, but this seems high.  Does this seem expected? For reference, I am testing it on the [linear algebra for java library](https://github.com/vkostyukov/la4j).

It seems the methods that take longer produce the warnings like the following:
```
[main] WARN sootup.callgraph.AbstractCallGraphAlgorithm - Could not find "java.lang.Object invokeBasic(java.lang.invoke.BoundMethodHandle)" in MethodHandle and in its superclasses
```
Is this a coincidence or a cause?

**Comment by JonasKlauke** [comment]
If your method contains JVM methods, they will cause a lot of edges in a call graph. 
If you analyze a Hello World program your callgraph will contain around 24.000 edges because of the single System.out.println call which takes around 8-10 minutes.

The warning just says that it could not find the method in the view. This can be caused by a missing jar as InputLocation or because the callgraph algorithm currently does not check if the found target is contained in the class. This will be fixed by #497 
This warning is just more likely to happen if the JVM is analyzed.

It might be faster to create the dispatches on your own because you are only interested in the outgoing edges of a single method in the call graph. You can look at the method resolveAllCallsFromSourceMethod in the class [AbstractCallGraphAlgorithm](https://github.com/soot-oss/SootUp/blob/develop/sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java)

**Comment by will-leeson** [comment]
I can definitely see that being expensive.

> It might be faster to create the dispatches on your own because you are only interested in the outgoing edges of a single method in the call graph

Thanks for the pointer. That did the trick. One question: Is there a reason `resolveCall` is protected and not public? Maybe I have a specific use, but I implemented a class which extends ClassHierarchyAnalysisAlgorithm and calls this function whenever I need to get the call edges. If you have a better solution, let me know.
